doc/architecture/timestamps_overview.adoc

```diff
@@ -0,0 +1,67 @@
+ifndef::include-only-once[]
+:root-path: ../
+include::{root-path}_config.adoc[]
+endif::[]
+= Timestamps
+
+== Simulation time
+
+`Simulation time` refers to the timestamp within a simulation environment.
+
+**Definition of zero time of simulation time**
+- Zero time is arbitrary but must be identical for all frames and corresponding messages (probably within one simulation).
+- Zero time does not need to coincide with the unix epoch.
+- It is recommended to use the start time of the simulation as zero time.
+
+**SensorData-specific timestamp**
+- Point in time that the sensor data message becomes available to the rest of the system (i.e. the driving functions).
+- Corresponds with the sending time / publish time.
+- Takes into account internal sensor processing latency.
+- Does not include bus communication delays (e.g. latency) that occur after the sensor output.
+
+**HostVehicleData-specific timestamp**
+- Time at which the host vehicle data message becomes available as a snapshot of the on-board network information.
+
+**Other top-level message timestamps**
+- GroundTruth, SensorView, TrafficCommand, TrafficUpdate, TrafficCommandUpdate, StreamingUpdate
+- The Timestamp coincides both with the notional simulation time the data applies to and with the time it was sent / published.
```

Comment:
How does timing work in a closed-loop co-simulation anyways? I am thinking about the timestamp of the TrafficUpdate, that gets fed back into the simulation engine / scenario player.

Let's say the scenario player publishes a SensorView with the timestamp 0.1 s.
A sensor model gets that, simulates the sensor output, which takes 100 ms, so it outputs a SensorData with timestamp 0.2 s.
That data gets processed by an object detection and path planning, which outputs to the actuator control and vehicle dynamics at 0.3 s.
This would also have some latency, for simplicity let's say also 100 ms. So the TrafficUpdate sent to the scenario player then has the timestamp 0.4.

Is my understanding generally correct? I have not really worked with closed-loop simulation much.
Can co-simulation platforms work with this delay?
Should we include an example like this in this document?

Reply:
I think it depends a lot on the simulation environment and how it's designed. You can do all sorts of things depending on what you are trying to do. But generally, I'd say that explaining how to do simulation time management is really beyond the scope of OSI and its documentation.

I'd rather keep this overview page small and concise. So far I have mainly copied and pasted existing definitions from the class reference documentation into this single document and tried to order/categorize it. The goal was to have a basis to reduce it to the essentials (e.g. extract similarities/differences of timestamps, identify missing definitions). The current state of this PR is just a draft, and nothing I want to merge as it is, as that just adds redundancy.

So I'd say the main point of a review right now would be to check if the categorization of timestamps is correct, if there are any timestamps missing or if their definitions are misleading/wrong/incomplete. We can then use this to update the definitions in the class reference documentation and reduce this section down to just explaining the different major categories/types of timestamps and making clear which fields belong to these categories. What do you think about this approach?

Reply:
Yes, I think this is a good approach. I agree that this page should be more of an overview and conceptual explanation. I just think that examples always help understanding definitions.